modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptNodeClientCodegen.java

```diff
@@ -128,4 +129,67 @@ public String getNpmRepository() {
     public void setNpmRepository(String npmRepository) {
         this.npmRepository = npmRepository;
     }
+
+    @Override
+    public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
+        Map<String, Object> result = super.postProcessAllModels(objs);
+
+        for (Map.Entry<String, Object> entry : result.entrySet()) {
+            Map<String, Object> inner = (Map<String, Object>) entry.getValue();
+            List<Map<String, Object>> models = (List<Map<String, Object>>) inner.get("models");
```

Comment:
At the end it has to throw an exception to interrupt code generation. I would consider adding ClassCastException handling to log the error and add @SuppressWarnings("unchecked") annotation to method. 
As a part of refactor I would consider introducing methods in parent class so it can be reused.

```
    @SuppressWarnings("unchecked")
    protected Map<String, Object> castMap(Object obj) {
        try {
            return (Map<String, Object>) obj;
        } catch (ClassCastException e) {
            LOGGER.error("...");
            throw e;
        }
    }

```

```
    @SuppressWarnings("unchecked")
    protected List<Map<String, Object>> castList(Object obj) {
        try {
            return (List<Map<String, Object>>) obj;
        } catch (ClassCastException e) {
            LOGGER.error("...");
            throw e;
        }
    }
```

```
    @SuppressWarnings("unchecked")
    protected CodegenModel castCodegenModel(Object obj) {
        try {
            return (CodegenModel) obj;
        } catch (ClassCastException e) {
            LOGGER.error("...");            
            throw e;
        }
    }

```